### PR TITLE
RFC#286 | Implement block let helper as syntax

### PR DESCRIPTION
**Pull request by locks**
I have started implementing [RFC#286](https://github.com/emberjs/rfcs/pull/286) so developers can start trying it out.

## Tasks
- [x] Add feature flag
- [x] Implement block `let`
- [x] Document implementation
- [x] Add tests

### CODE REVIEW

File: packages/ember-glimmer/lib/syntax/let.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  `content` and `title` properties, and a `user` object with `firstName`
+  and `lastName` properties.
+
+  We are going to call the `my-post` component, passing a title with is
+  the title of the post suffixed with the name of the blog, the content
+  of the post, and a series of options defined in-place.
+
+  ```handlebars
+  {{#let
+      (concat post ' | The Ember.js Blog' )
```
Comment (GavinJoyce) [inline packages/ember-glimmer/lib/syntax/let.ts]: nitpick, `(concat post ' | The Ember.js Blog')`

File: packages/ember-glimmer/tests/integration/syntax/let-test.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+import { EMBER_TEMPLATE_BLOCK_LET_HELPER } from 'ember/features';
+
+if (EMBER_TEMPLATE_BLOCK_LET_HELPER) {
+  moduleFor('Syntax test: {{#let as}}', class extends RenderingTest  {
+
+    templateFor({ cond, truthy, falsy }) {
+      return `{{#let ${cond} as |test|}}${truthy}{{else}}${falsy}{{/let}}`;
+    }
+
+    ['@test keying off of `undefined` does not render']() {
```
Comment (GavinJoyce) [inline packages/ember-glimmer/tests/integration/syntax/let-test.js]: I find the description a little hard to parse. Perhaps `@test a key with an undefined value renders nothing`?

Comment (locks) [reply packages/ember-glimmer/tests/integration/syntax/let-test.js]: Oops, the tests were adapted from `{{with}}` and this should not actually pass 🤔 .

Comment (locks) [reply packages/ember-glimmer/tests/integration/syntax/let-test.js]: It passes because `undefined` renders as ``, hah. Adapted the test for `{{let}}`.

File: packages/ember-glimmer/lib/syntax/let.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        theme="high-contrast"
+        enableComments=true
+      )
+      as |title username options|
+  }}
+    {{my-post title=title content=blog.content options=options}}
+  {{/let}}
+  ```
+
+  @method let
```
Comment (locks) [inline packages/ember-glimmer/lib/syntax/let.ts]: I am not sure what other annotations to put here

Comment (locks) [reply packages/ember-glimmer/lib/syntax/let.ts]: cc @toddjordan

File: packages/ember-glimmer/lib/syntax/let.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  them out as block params.
+
+  This allows the developer to introduce shorter names for certain computations
+  in the template.
+
+  This is especially useful if you are passing properties to a component
+  that receives a lot of options and you want to clean up the invocation.
+
+  For the following example, the template receives a `post` object with
+  `content` and `title` properties, and a `user` object with `firstName`
```
Comment (rwjblue) [inline packages/ember-glimmer/lib/syntax/let.ts]: I don't see a `user` object passed in the example below, maybe that was from a prior iteration?

File: packages/ember-glimmer/lib/syntax/let.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  and `lastName` properties.
+
+  We are going to call the `my-post` component, passing a title with is
```
Comment (rwjblue) [inline packages/ember-glimmer/lib/syntax/let.ts]: s/with is/which is/

**Review by rwjblue — APPROVED** [review]
